Source/RapyutaSimulationPlugins/Private/Core/RRBaseActor.cpp

```diff
@@ -46,16 +45,15 @@ void ARRBaseActor::PreInitializeComponents()
     // SETUP + CONFIGURE ESSENTIAL GAME & SIM COMMON OBJECTS
     // NOTE: These are used ONLY for dynamic runtime actors.
     // In child BP actors, except for GameSingleton, other Game framework entities maybe not available yet
-    GameMode = URRCoreUtils::GetGameMode<ARRGameMode>(this);
+#if RAPYUTA_USE_SCENE_DIRECTOR
     GameState = URRCoreUtils::GetGameState<ARRGameState>(this);
-
-    if (IsNetMode(NM_Standalone))
+    check(GameState)
+#endif
+    GameSingleton = URRGameSingleton::Get();
+    if(GameSingleton==nullptr)
     {
-        PlayerController = URRCoreUtils::GetPlayerController<ARRPlayerController>(SceneInstanceId, this);
+         UE_LOG_WITH_INFO_NAMED(LogRapyutaCore, Display, TEXT("GameSingleton is not child class of URRGameSingleton. Cannot use Asset loading."));
     }
```

Comment:
As far as I checked only GameState and GameSingleton are used.
I think it is better to remove other pointers until those become needed again to avoid confusion

Reply:
So the idea of `RapyutaSimPlugins` is to provide a suite of handy `RR~` actors that should come along but with minimal coupling. I suppose the confusion comes from the fact that we used `check` for debugging, which is now already removed now in this PR. In fact,
   * `GameMode` is very essential eg used [here](https://github.com/rapyuta-robotics/RapyutaSimulationPlugins/blob/devel/Source/RapyutaSimulationPlugins/Private/Core/RRProceduralMeshComponent.cpp#L122), as of `ROS2GameMode`, which contains SimState.
-> Having a convenient pointer here would facilitate us to quickly fetch `SimState` from any `RRBaseActor` like [here](https://github.com/rapyuta-robotics/RapyutaSimulationInternal/blob/RuntimePhysicsPlatter/Source/RapyutaSimInternal/Private/Forklift/RRPlatter.cpp#L940), without having to call the expensive `GetAllActorsOfClass` everytime we need to fetch it.
   * `PlayerController` is mostly used for DataSynth app.

Anyway, they are fetched here mostly for convenience to be used later, so we could just have to do a simple cast in client project/plugin without having to fetch them again by `GetWorld()->GetAuthGameMode()`

Reply:
So the idea of `RapyutaSimPlugins` is to provide a suite of handy `RR~` actors that should come along but with minimal coupling. I suppose the confusion comes from the fact that we used `check` for debugging, which is now already removed in this PR. In fact,
   * `GameMode` is very essential eg used [here](https://github.com/rapyuta-robotics/RapyutaSimulationPlugins/blob/devel/Source/RapyutaSimulationPlugins/Private/Core/RRProceduralMeshComponent.cpp#L122), as of `ROS2GameMode`, which contains SimState.
-> Having a convenient pointer here would facilitate us to quickly fetch `SimState` from any `RRBaseActor` like [here](https://github.com/rapyuta-robotics/RapyutaSimulationInternal/blob/RuntimePhysicsPlatter/Source/RapyutaSimInternal/Private/Forklift/RRPlatter.cpp#L940), without having to call the expensive `GetAllActorsOfClass` everytime we need to fetch it.
   * `PlayerController` is mostly used for DataSynth app.

Anyway, they are fetched here mostly for convenience to be used later, so we could just have to do a simple cast in client project/plugin without having to fetch them again by `GetWorld()->GetAuthGameMode()`

Reply:
OK then,
I changed variable name for clarification

Reply:
Why do  you needs to call `GetAllActorsOfClass` to fetch SimState? Can' you use GetGameMode()->SimState work?

Reply:
Yeap that should be the correct way to fetch it, though sometimes not always followed like [here](https://github.com/rapyuta-robotics/RapyutaSimulationInternal/blob/devel/Source/RapyutaSimInternal/Private/Human/RRHumanROS2Interface.cpp#L126).
Essentially, repeated `GetGame~()` call every time it needs to be used should be avoided for reasons:
* It is a UE-api call, so we don't have full control over the overheads it takes. (like it typically involves `GetWorld()` which is not a trivial call at all).
* Even if we know it's cheap to call, it's always cheaper to fetch it once only (except it is changed at runtime but Game~ actors do not) and use during Sim run, which is a common practice also applied to fetching UE-asset resource.

Reply:
- Why you didn't follow correct way there?
- Are there any documentation that shows `GetWorld()` is not a trivial call? I think it is difficult to know all api which is slow but we can have list of those function.

Reply:
Yeah,
* It is not my code, it is my observation of the inefficient use of the API, which happens when the class does not hold a quick pointer to `RRGame~`.
* If you look at UE-builtin `GetWorld()/GetGame~()` implementation, it does a couple of queries and lookups, which might be or not be much of overheads but apparently, it's always cheaper if we already have a pointer `UWorld*` (of course as long as it is not updated during Sim run). That practice is applied to any type of pointer referencing in general, as the less referencing is done, less lookup work for the processor. 
* Anyway, since they are UE-apis thus better not to use them too expediently since their implementation are prone to be updated, which is also why we have Utils functions instead of direct calls to UE-apis for:
+ To reduce duplication.
+ To reduce coupling like if they are deprecated, we just need to update Util function instead of going to all places in oursourcebase to update it.